src/core/server/docs/kib_core_reviewing_so_type_pr.mdx

```diff
@@ -0,0 +1,78 @@
+---
+id: kibCoreReviewingSoPr
+slug: /kibana-dev-docs/review/reviewing-so-pr
+title: Reviewing SavedObject PRs
+description: How to review PRs that changes savedObjects registration
+date: 2022-09-30
+tags: ['kibana','dev', 'contributor', 'api docs']
+---
+
+# Reviewing PRs that change Saved Object types
+
+## How does automatic review assignment work when SO types are changed?
+
+PRs modifying / adding / deleting any SO type registration will be flagged by the integration
+test located at `src/core/server/integration_tests/saved_objects/migrations/check_registered_types.test.ts`.
+
+This test will fail any time a change is performed on an SO type registration that could risk having an impact on 
+upgrades/migrations, and will force the PR's author to update the test's snapshot, which will trigger a review 
+from the Core team.
+
+## What and how to review?
+
+Reviews will be triggered when one, or more, of these scenarios occur:
+- a new type was added (registered)
+- a type was removed (is no longer registered)
+- a new migration function was added
+- a mapping change was performed
+- a new validation schema was added
+- an `excludeOnUpgrade` function was added or removed
+
+Note: reviews will **not** automatically be triggered in these scenarios:
+- an existing migration function is changed
+- an existing validation schema is changed
+- an existing `excludeOnUpgrade` function is changed
+
+### A new type was added
+
+We have another integration test detecting this scenario (`src/core/server/integration_tests/saved_objects/migrations/type_registrations.test.ts`)
+
+In that scenario, we should:
+- check the initial mappings of the type to spot potential issues:
+  - fields being defined explicitly in the mappings but not directly used for search
+  - overall amount of fields is high
+  - use of `dynamic: true` (this can lead to mapping explosions)
+- check if the type is registered as `hidden: true` and encourage to do so otherwise. 
+  - this avoids polluting the global SO HTTP APIs with another type, and instead requires plugin developers to build 
+    their own HTTP APIs to access this type of SO if they truly need to.
+
+### A type was removed
+
+The integration test mentioned in the previous section also detects those scenarios. 
+
+Here, we need to check:
+- that `REMOVED_TYPES` (`packages/core/saved-objects/core-saved-objects-migration-server-internal/src/core/unused_types.ts`) was properly updated 
+  - (but, again, the integration test will fail otherwise)
+- that no other existing SO types may be referencing this type
+  - can't really be automated, will likely need to ask the owning team directly
+
+### A new migration function was added
+
+- Review the migration function
+    - owners are supposed to do it, but an additional review never hurts
+- Make sure that the migration function is properly tested
+  - by unit tests
+  - and, ideally, by integration tests using 'real' data
```

Comment:
we can link to our migration testing docs [dev_docs/tutorials/testing_plugins.mdx](https://github.com/elastic/kibana/blob/es-socket-config/dev_docs/tutorials/testing_plugins.mdx#L796)